include/deal.II/base/tensor.h

```diff
@@ -2299,15 +2213,24 @@ Number determinant (const Tensor<2,dim,Number> &t)
         for (unsigned int j=0; j<dim-1; ++j)
           minor[i][j] = t[i][j<k ? j : j+1];
 
-      const Number cofactor = std::pow (-1., static_cast<Number>(k+1)) *
-                              determinant (minor);
+      const Number cofactor = ((k % 2 == 0) ? -1. : 1.) * determinant(minor);
 
       det += t[dim-1][k] * cofactor;
     }
 
-  return std::pow (-1., static_cast<Number>(dim)) * det;
+  return ((dim % 2 == 0) ? 1. : -1.) * det;
```

Comment:
Did you make sure that for dim=1,2,3 you get code that is as efficient as the original specializations?